iDiff/utils/image_prep_utils.go

```diff
@@ -129,32 +89,47 @@ type CloudPrepper struct {
 }
 
 func (p CloudPrepper) ImageToFS() (string, error) {
-	// check client compatibility with Docker API
-	valid, err := ValidDockerVersion()
+	URLPattern := regexp.MustCompile("^.+/(.+(:.+){0,1})$")
+	URLMatch := URLPattern.FindStringSubmatch(p.Source)
+	path := strings.Replace(URLMatch[1], ":", "", -1)
```

Comment:
Added a comment.  The first line describes the patterns being looked for such that the second line puts the pieces that fit those patterns into a list ( [ entire string, everything after "/", everything after : after / ].  So doing URLMatch[1] gives you the image name and tag and we take out the colon to be able to use that path for saving.